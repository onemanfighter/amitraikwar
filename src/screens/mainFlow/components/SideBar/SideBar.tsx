import { MoonIcon, SearchIcon, SunIcon } from '@assets';
import {
  Box,
  Button,
  HStack,
  IconButton,
  Text,
  Tooltip,
  useColorMode,
} from '@chakra-ui/react';
import { useTranslation } from 'react-i18next';
import { Link } from 'react-router-dom';
import TooltipButtons from './components/TooltipButtons';
import Content from './components/Content';
import HeaderText from './components/HeaderText';

const SideBar = () => {
  return (
    <Box position={'sticky'} top={'0'} marginBottom={8}>
      <HeaderText />
      <TooltipButtons />
      <Content />
    </Box>
  );
};

export default SideBar;
