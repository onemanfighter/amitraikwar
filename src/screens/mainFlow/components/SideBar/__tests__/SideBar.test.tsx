import { render } from '@testing-library/react';
import SideBar from '../SideBar';
import { ThemeProvider } from '@providers';
import { BrowserRouter } from 'react-router-dom';

jest.mock('react-social-icons', () => ({
  SocialIcon: ({ url }: { url: string }) => <div>{url}</div>,
}));

describe('Side bar', () => {
  it('should renders correctly', () => {
    const { container } = render(
      <ThemeProvider>
        <BrowserRouter>
          <SideBar />
        </BrowserRouter>
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });
});
