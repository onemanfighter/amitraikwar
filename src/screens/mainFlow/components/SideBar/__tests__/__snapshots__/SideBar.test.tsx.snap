// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Side bar should renders correctly 1`] = `
<div>
  <script
    id="chakra-script"
  >
    !(function(){try{var a=function(c){var v="(prefers-color-scheme: dark)",h=window.matchMedia(v).matches?"dark":"light",r=c==="system"?h:c,o=document.documentElement,s=document.body,l="chakra-ui-light",d="chakra-ui-dark",i=r==="dark";return s.classList.add(i?d:l),s.classList.remove(i?l:d),o.style.colorScheme=r,o.dataset.theme=r,r},n=a,m="light",e="chakra-ui-color-mode",t=localStorage.getItem(e);t?a(t):localStorage.setItem(e,a(m))}catch(a){}})();
  </script>
  <div
    class="css-1bjuvht"
  >
    <a
      class="chakra-button css-ik3kil"
      href="/"
    >
      Amit Raikwar
    </a>
    <div
      class="css-xy3nmn"
    >
      <button
        class="chakra-button css-bgkogr"
        type="button"
      >
        Download CV
      </button>
      <button
        aria-label="3D toggle"
        class="chakra-button css-ysx01z"
        type="button"
      >
        <svg
          class="w-6 h-6"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m21 7.5-9-5.25L3 7.5m18 0-9 5.25m9-5.25v9l-9 5.25M3 7.5l9 5.25M3 7.5v9l9 5.25m0-9v9"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </button>
      <button
        aria-label="Search"
        class="chakra-button css-ysx01z"
        type="button"
      >
        <svg
          class="w-6 h-6"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m21 21-5.197-5.197m0 0A7.5 7.5 0 1 0 5.196 5.196a7.5 7.5 0 0 0 10.607 10.607Z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </button>
      <button
        aria-label="Toggle dark mode"
        class="chakra-button css-ysx01z"
        type="button"
      >
        <svg
          class="w-6 h-6"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21.752 15.002A9.72 9.72 0 0 1 18 15.75c-5.385 0-9.75-4.365-9.75-9.75 0-1.33.266-2.597.748-3.752A9.753 9.753 0 0 0 3 11.25C3 16.635 7.365 21 12.75 21a9.753 9.753 0 0 0 9.002-5.998Z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </button>
    </div>
    <div
      class="css-i9axev"
    >
      <div
        class="css-12f1a2y"
      >
        <img
          class="chakra-image css-1hke5j3"
          loading="lazy"
          src=""
        />
      </div>
      <div
        class="css-16mslk0"
      >
        <span
          class="index-module_type__E-SaG"
          style="font-size: 1.3em; font-weight: bold; font-family: monospace; text-align: center;"
        />
      </div>
      <div
        class="css-1n7u47g"
      >
        <div
          class="css-1t5din3"
        >
          <button
            aria-label="+91-8756366219"
            class="chakra-button css-r1mtb4"
            type="button"
          >
            <svg
              class="w-6 h-6"
              fill="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10.5 18.75a.75.75 0 0 0 0 1.5h3a.75.75 0 0 0 0-1.5h-3Z"
              />
              <path
                clip-rule="evenodd"
                d="M8.625.75A3.375 3.375 0 0 0 5.25 4.125v15.75a3.375 3.375 0 0 0 3.375 3.375h6.75a3.375 3.375 0 0 0 3.375-3.375V4.125A3.375 3.375 0 0 0 15.375.75h-6.75ZM7.5 4.125C7.5 3.504 8.004 3 8.625 3H9.75v.375c0 .621.504 1.125 1.125 1.125h2.25c.621 0 1.125-.504 1.125-1.125V3h1.125c.621 0 1.125.504 1.125 1.125v15.75c0 .621-.504 1.125-1.125 1.125h-6.75A1.125 1.125 0 0 1 7.5 19.875V4.125Z"
                fill-rule="evenodd"
              />
            </svg>
          </button>
          <div
            class="css-o99mrc"
          >
            <p
              class="chakra-text css-itvw0n"
            >
              Phone
            </p>
            <p
              class="chakra-text css-avjjns"
            >
              +91-8756366219
            </p>
          </div>
        </div>
        <div
          class="css-1t5din3"
        >
          <button
            aria-label="raikwar.amit.1603@gmail.com"
            class="chakra-button css-r1mtb4"
            type="button"
          >
            <svg
              class="w-6 h-6"
              fill="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1.5 8.67v8.58a3 3 0 0 0 3 3h15a3 3 0 0 0 3-3V8.67l-8.928 5.493a3 3 0 0 1-3.144 0L1.5 8.67Z"
              />
              <path
                d="M22.5 6.908V6.75a3 3 0 0 0-3-3h-15a3 3 0 0 0-3 3v.158l9.714 5.978a1.5 1.5 0 0 0 1.572 0L22.5 6.908Z"
              />
            </svg>
          </button>
          <div
            class="css-o99mrc"
          >
            <p
              class="chakra-text css-itvw0n"
            >
              Email
            </p>
            <p
              class="chakra-text css-avjjns"
            >
              raikwar.amit.1603@gmail.com
            </p>
          </div>
        </div>
        <div
          class="css-1t5din3"
        >
          <button
            aria-label="Pune, India"
            class="chakra-button css-r1mtb4"
            type="button"
          >
            <svg
              class="w-6 h-6"
              fill="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="m11.54 22.351.07.04.028.016a.76.76 0 0 0 .723 0l.028-.015.071-.041a16.975 16.975 0 0 0 1.144-.742 19.58 19.58 0 0 0 2.683-2.282c1.944-1.99 3.963-4.98 3.963-8.827a8.25 8.25 0 0 0-16.5 0c0 3.846 2.02 6.837 3.963 8.827a19.58 19.58 0 0 0 2.682 2.282 16.975 16.975 0 0 0 1.145.742ZM12 13.5a3 3 0 1 0 0-6 3 3 0 0 0 0 6Z"
                fill-rule="evenodd"
              />
            </svg>
          </button>
          <div
            class="css-o99mrc"
          >
            <p
              class="chakra-text css-itvw0n"
            >
              Location
            </p>
            <p
              class="chakra-text css-avjjns"
            >
              Pune, India
            </p>
          </div>
        </div>
        <div
          class="css-1t5din3"
        >
          <button
            aria-label="16th March 1998"
            class="chakra-button css-r1mtb4"
            type="button"
          >
            <svg
              class="w-6 h-6"
              fill="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m15 1.784-.796.795a1.125 1.125 0 1 0 1.591 0L15 1.784ZM12 1.784l-.796.795a1.125 1.125 0 1 0 1.591 0L12 1.784ZM9 1.784l-.796.795a1.125 1.125 0 1 0 1.591 0L9 1.784ZM9.75 7.547c.498-.021.998-.035 1.5-.042V6.75a.75.75 0 0 1 1.5 0v.755c.502.007 1.002.021 1.5.042V6.75a.75.75 0 0 1 1.5 0v.88l.307.022c1.55.117 2.693 1.427 2.693 2.946v1.018a62.182 62.182 0 0 0-13.5 0v-1.018c0-1.519 1.143-2.829 2.693-2.946l.307-.022v-.88a.75.75 0 0 1 1.5 0v.797ZM12 12.75c-2.472 0-4.9.184-7.274.54-1.454.217-2.476 1.482-2.476 2.916v.384a4.104 4.104 0 0 1 2.585.364 2.605 2.605 0 0 0 2.33 0 4.104 4.104 0 0 1 3.67 0 2.605 2.605 0 0 0 2.33 0 4.104 4.104 0 0 1 3.67 0 2.605 2.605 0 0 0 2.33 0 4.104 4.104 0 0 1 2.585-.364v-.384c0-1.434-1.022-2.7-2.476-2.917A49.138 49.138 0 0 0 12 12.75ZM21.75 18.131a2.604 2.604 0 0 0-1.915.165 4.104 4.104 0 0 1-3.67 0 2.605 2.605 0 0 0-2.33 0 4.104 4.104 0 0 1-3.67 0 2.605 2.605 0 0 0-2.33 0 4.104 4.104 0 0 1-3.67 0 2.604 2.604 0 0 0-1.915-.165v2.494c0 1.035.84 1.875 1.875 1.875h15.75c1.035 0 1.875-.84 1.875-1.875v-2.494Z"
              />
            </svg>
          </button>
          <div
            class="css-o99mrc"
          >
            <p
              class="chakra-text css-itvw0n"
            >
              Birthday
            </p>
            <p
              class="chakra-text css-avjjns"
            >
              16th March 1998
            </p>
          </div>
        </div>
      </div>
      <div
        class="css-n0xb99"
      >
        <p
          class="chakra-text css-1xs2h3i"
        >
          Software Engineer 3 @ RSLabs
        </p>
      </div>
      <div
        class="css-11avr8y"
      >
        <hr
          aria-orientation="horizontal"
          class="chakra-divider css-svjswr"
        />
        <div
          class="css-jqz8zx"
        >
          Social links
        </div>
      </div>
      <div
        class="css-17lwykd"
      >
        <div
          class="css-pocmpu"
        >
          <div>
            https://in.linkedin.com/in/amitrai1603
          </div>
        </div>
        <div
          class="css-pocmpu"
        >
          <div>
            https://github.com/onemanfighter
          </div>
        </div>
        <div
          class="css-pocmpu"
        >
          <div>
            https://github.com/raikwaramit
          </div>
        </div>
        <div
          class="css-pocmpu"
        >
          <div>
            https://amitraikwar.medium.com/
          </div>
        </div>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
